pkg/specgen/specgen.go

```diff
@@ -594,28 +597,37 @@ var (
 )
 
 // NewSpecGenerator returns a SpecGenerator struct given one of two mandatory inputs
-func NewSpecGenerator(arg string, rootfs bool) *SpecGenerator {
+func NewSpecGenerator(arg string, rootfs bool) (*SpecGenerator, error) {
```

Comment:
I also argue here, that this is a breaking change. While we only consider pkg/bindings stable the container create binding needs a specgen as input. I assume that most people use this function to create one.
I rather not break them. We should move the validation down the stack to prevent this.